src/_pytest/logging.py

```diff
@@ -595,6 +602,15 @@ def __init__(self, config: Config) -> None:
         )
         self.log_cli_handler.setFormatter(log_cli_formatter)
 
+        if config.option.suppress_logger:
+            self._suppress_loggers()
+
+    def _suppress_loggers(self) -> None:
+        logger_names_to_suppress = set(self._config.option.suppress_logger)
+        for name in logger_names_to_suppress:
+            logger = logging.getLogger(name)
+            logger.addFilter(lambda _: 0)
```

Comment:
Instead of editing handlers and/or loggers; I opted for adding a filter to the logger directly.
@ankostis thoughts?

From logging v3.2 filters can be Callables ([docs](https://docs.python.org/3/library/logging.html#filter-objects)) instead of classes.

Reply:
Filters seems like a nice solution.

Wouldn't the user also expect the [`logger.propagate`](https://docs.python.org/3/library/logging.html#logging.Logger.propagate) to become `False`?

I guess an alternative would be to stop logger's propagation and remove all of its handlers.

Reply:
(further elaboration on my message above)

The logger hierarchy needs further attention;  from this sentence in [`propagate` docstring](https://docs.python.org/3/library/logging.html#logging.Logger.propagate):

> Messages are passed directly to the ancestor loggers’ handlers - neither the level nor filters of the ancestor loggers in question are considered.

... confirms that the filter won't prevent log-records from children-loggers to reach ancestor/root ones -  so `propagate = False` is definitely needed. 

To have a super correct behavior, the tree of loggers must be traversed to identify all logger-children and disable them with a filter or by removing all their handlers - but we can defer that, and assume the user will see the any logs originating from child logger(s) and append also those in the new CLI option.

Reply:
Indeed, didn't read the 2nd point you raised - and the tests confirm it.
I guess the docstrings need some attention here.

But checking the code, i noticed there is the [`Logger.disabled` attribute](https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L1507),
which does exactly what we want, no?

Reply:
Indeed! I changed the code to use `logger.disabled` instead. What do you think?

Reply:
+1